### ISSUE TITLE
Changelog and release notes for 0.0.10

### ISSUE THREAD

**Issue by dexX7** [OmniLayer/omnicore#96]
It would be great, if the changes since the last release would be documented.

I think there is no need to go into very much detail for the obvious parts, but there are a few points, which are good to know:

- new transaction types for token exchange
- rebranding (file names!)
- OP_RETURN data embedding
- new API calls
- new configuration options
- debug log customization

**Comment by zathras-crypto** [comment]
Changelogs have been requested by integrators before, and I thoroughly agree we need one - we're probably talking on the order of 4 digit commit counts so obviously not feasible to stipulate every commit, but agree stepping through and highlighting everything substantial should be done.

**Comment by dexX7** [comment]
After reading through the PR titles:

- Upgrade code base to Bitcoin Core 0.10.2
- Select only coins, which can be spent 
- Fix shutdown issue after running Boost tests
- Finalize MetaDEx logic
- MetaDEx user interface
- Support creation of any (enabled) Omni transaction via RPC
- Class C/OP_RETURN encoded transactions
- Report initial parsing progress during startup
- Handle shutdown requests during initial parsing
- Integrate Travis CI and OmniJ into the test and workflow
- Allow debug levels to be specified without needing to recompile
- Set `txindex=1` flag during startup, if confirmed by user
- Various code cleanups and improved code documentation
- Resolve and refine several `LOCKs`
- Support concurrent RPC queries
- Show script verification errors in `signrawtransaction` result
- Enable pay-to-script-hash support in send dialog
- Return all available information via `validateaddress`
- Indicate "pending" status of outgoing transactions
- Log every invalid processing of Omni transactions
- Rebrand project to Omni Core
- Rename files to `omnicore-qt`, `omnicored`, `omnicore-cli`, ...
- Fully support cross-plattform, and deterministic, building
- Unify and improve RPC help descriptions
- Fix shutdown issue after declining to reindex blockchain

I'm not sure, if this is very helpful, but probably a start.

**Comment by dexX7** [comment]
I'm going to keep the list updated, and add new entries, once the PRs are merged.

But let's discuss the general plan for the future: while a list with PR titles isn't bad, it's not necessarily very useful, given that titles are usually pretty short and consist mostly of keywords. I don't really see a great alternative, except using commit titles, and no PR titles.

Alternatively we may create a new document, which should be updated, if a new PR is proposed/merged, sort of like maintaining the documentation in parallel?

**Comment by zathras-crypto** [comment]
> I'm not sure, if this is very helpful, but probably a start.

Hugely helpful :)

> For the future: please let's create PRs with only one, single, purpose, and clear titles, which can be copied 1 to 1 into the next chagelog, hehe.

Agreed - I'll try to "clean up" my work methods as I do have a habit of working on several different things at the same time and basing one new thing on top of another new thing - I'll try to be a bit better at branching :)

> Alternatively we may create a new document, which should be updated, if a new PR is proposed/merged, sort of like maintaining the documentation in parallel?

Could we not simply require that every PR requires a "changelog description" text which is more descriptive that just a title and then use that to maintain a changelog?  Just thinking out loud there :)

**Comment by dexX7** [comment]
> Could we not simply require that every PR requires a "changelog description" text which is more descriptive that just a title and then use that to maintain a changelog? Just thinking out loud there :)

Haha, yes! Ideally the PR text includes a description, which could be read isolated (i.e. in a changelog/documentation), and should fully describe *what the PR does* (not necessarily *how* [code wise]). We should note that code discussion, and notes, and anything that is not descriptive-change-text should be posted in the second post. PR texts should be updated, if the PR was updated.

Sounds good? I have a bit a hard time to put this is more "user faced" words, but we should add a `CONTRIBUTION.md` file. [See here](https://github.com/docker/docker/issues/new) the nice "hint", if one creates a new issue, and [here](https://github.com/docker/docker/blob/master/CONTRIBUTING.md) the very awesome guidelines.

**Comment by dexX7** [comment]
I started to tag pull requests with labels (other than the status indicating labels we already use) to make it easier to categorize changes.

Please use labels, whenever you stumble upon something that is missing or needs to be updated. If no appropriate label is available, feel free to add new ones.. :)

**Comment by dexX7** [comment]
This also raises the question whether we should add labels for old pull requests.
